### PR TITLE
[doc] Contribute the ADR for the support of the selection dialog

### PR DESCRIPTION
**Pull request by sbegaudeau**

### CODE REVIEW

File: doc/adrs/022_add_support_for_selection_dialog.adoc [hunk 1]
```diff
@@ -0,0 +1,129 @@
+= ADR-022 - Add support for selection dialog
+
+== Context
+
+We want to support Sirius RCP's `Select Model Element Variable` for the tools of the diagram.
+
+== Decision
+
+We will only support one and only one `Select Model Element Variable` per tool and we will start with the Node Creation Tool for the moment.
+
```
Comment (AxelRICHARD) [inline doc/adrs/022_add_support_for_selection_dialog.adoc:10]: The `Select Model Element Variable` will only support single selection and list display (no tree display) for the moment.

File: doc/adrs/022_add_support_for_selection_dialog.adoc [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+In a similar fashion as for the diagrams, a type provider will have to be created (like `DiagramTypesProvider`) in order to register our new representation in the GraphQL schema.
+We will need to be careful with the GraphQL annotation on our POJOs (Selection, SelectionObject, SelectionDescription, etc).
+A DataFetcher will also have to be created for the representation, the `SubscriptionSelectionEventDataFetcher` in a similar fashion as the `SubscriptionTreeEventDataFetcher`.
+In order to create the GraphQL subscription, an event processor and its factory will be needed just like the Tree representation for example.
+
+
+=== Execution
+
+When a tool will be executed, the frontend will give back to the backend the id of the selected element.
```
Comment (AxelRICHARD) [inline doc/adrs/022_add_support_for_selection_dialog.adoc:117]: ... of the selected element of the selection dialog.

File: doc/adrs/022_add_support_for_selection_dialog.adoc (hunk 1 above)
Comment (sbegaudeau) [reply doc/adrs/022_add_support_for_selection_dialog.adoc:10]: Added to the document

File: doc/adrs/022_add_support_for_selection_dialog.adoc (hunk 2 above)
Comment (sbegaudeau) [reply doc/adrs/022_add_support_for_selection_dialog.adoc:117]: Added to the document

**Review by gcoutable — APPROVED** [review]

**Pull request merged**